--- CMakeLists.txt
+++ CMakeLists.txt
@@ -92,6 +92,8 @@
   SET(sys_c "sys_ansi.c")
   SET(sys_h "sys_ansi.h")
   SET(sys_compat_c "sys_compat_unix.c")
+
+  OPTION(BUILD_SHARED_LIBS "Build shared libpano13.so library" OFF)
 ENDIF(WIN32)
 
 OPTION(SUPPORT_JAVA_PROGRAMS "Set this to support for the Java programs ptpicker and pteditor" OFF)
@@ -263,16 +265,20 @@
 include_directories(${CMAKE_SOURCE_DIR})
 
 if(NOT WIN32)
-  # Create and install shared library on non-windows
-  add_library(pano13 SHARED ${pano13_hdr} ${pano13_res} ${pano13_src})
-  target_link_libraries(pano13 ${_common_libs})
-  set_target_properties(pano13 PROPERTIES VERSION "3.0.0" SOVERSION "3")
-  install(TARGETS pano13 LIBRARY DESTINATION ${MYLIB} COMPONENT libraries)
-  set(_pano_target pano13)
-  #create and install static library
+  # Create and install static library on non-windows
   add_library(pano13a STATIC ${pano13_hdr} ${pano13_res} ${pano13_src})
   set_target_properties(pano13a PROPERTIES OUTPUT_NAME pano13)
   install(TARGETS pano13a ARCHIVE DESTINATION ${MYLIB} COMPONENT libraries)
+  if (BUILD_SHARED_LIBS)
+    # Create and install shared library on non-windows
+    add_library(pano13 SHARED ${pano13_hdr} ${pano13_res} ${pano13_src})
+    target_link_libraries(pano13 ${_common_libs})
+    set_target_properties(pano13 PROPERTIES VERSION "3.0.0" SOVERSION "3")
+    install(TARGETS pano13 LIBRARY DESTINATION ${MYLIB} COMPONENT libraries)
+    set(_pano_target pano13)
+  else(BUILD_SHARED_LIBS)
+    set(_pano_target pano13a)
+  endif(BUILD_SHARED_LIBS)
 else()
   IF(BUILD_DLL)
     # Create and install shared library
@@ -329,8 +335,6 @@
 
 add_subdirectory(man)
 add_dependencies(${_pano_target} man)
-add_subdirectory(tools)
-add_dependencies(tools ${_pano_target})
 
 macro(setstripped _varname _value)
   string(STRIP ${_value} _v)
