--- Makefile	2023-11-02 11:03:29.756289853 +0100
+++ Makefile	2023-11-02 11:49:43.758264138 +0100
@@ -37,7 +37,7 @@
 SKIPHEADERS = compat/w32pthreads.h
 
 # first so "all" becomes default target
-all: all-yes
+all: redact all-yes
 
 include $(SRC_PATH)/tools/Makefile
 include $(SRC_PATH)/ffbuild/common.mak
@@ -194,5 +194,9 @@
 # so this saves some time on slow systems.
 .SUFFIXES:
 
-.PHONY: all all-yes alltools build check config testprogs
-.PHONY: *clean install* uninstall*
+.PHONY: all all-yes alltools build check config testprogs redact
+.PHONY: *cl
+
+redact::
+	sed --regexp-extended 's:(--[^=]+=)(/[^/ ]+)*/:\1/redacted/:g' \
+		-i config.h
