diff --git emscripten_toolchain/emar.sh emscripten_toolchain/emar.sh
index e4279f1..7e73539 100755
--- emscripten_toolchain/emar.sh
+++ emscripten_toolchain/emar.sh
@@ -1,5 +1,3 @@
 #!/bin/bash
 
-source $(dirname $0)/env.sh
-
-exec python3 $EMSCRIPTEN/emar.py "$@"
+exec "${PYTHON:-python3}" "${EMSCRIPTEN}/emar.py" "$@"
diff --git emscripten_toolchain/emcc.sh emscripten_toolchain/emcc.sh
index 7f3699b..95b3200 100755
--- emscripten_toolchain/emcc.sh
+++ emscripten_toolchain/emcc.sh
@@ -1,5 +1,3 @@
 #!/bin/bash
 
-source $(dirname $0)/env.sh
-
-exec python3 $EMSCRIPTEN/emcc.py "$@"
+exec "${PYTHON:-python3}" "${EMSCRIPTEN}/emcc.py" "$@"
diff --git emscripten_toolchain/emcc_link.sh emscripten_toolchain/emcc_link.sh
index 24d806d..6163b9b 100755
--- emscripten_toolchain/emcc_link.sh
+++ emscripten_toolchain/emcc_link.sh
@@ -1,5 +1,3 @@
 #!/bin/bash
 
-source $(dirname $0)/env.sh
-
-exec python3 $(dirname $0)/link_wrapper.py "$@"
+exec "${PYTHON:-python3}" "${EXT_BUILD_ROOT:-.}/external/emsdk/emscripten_toolchain/link_wrapper.py" "$@"
diff --git emscripten_toolchain/link_wrapper.py emscripten_toolchain/link_wrapper.py
index 1e26bde..aa1ca90 100644
--- emscripten_toolchain/link_wrapper.py
+++ emscripten_toolchain/link_wrapper.py
@@ -21,22 +21,11 @@
 import subprocess
 import sys
 
-# Only argument should be @path/to/parameter/file
-assert sys.argv[1][0] == '@', sys.argv
-param_filename = sys.argv[1][1:]
-param_file_args = [l.strip() for l in open(param_filename, 'r').readlines()]
-
-# Re-write response file if needed.
-if any(' ' in a for a in param_file_args):
-  new_param_filename = param_filename + '.modified'
-  with open(new_param_filename, 'w') as f:
-    for param in param_file_args:
-      if ' ' in param:
-        f.write('"%s"' % param)
-      else:
-        f.write(param)
-      f.write('\n')
-  sys.argv[1] = '@' + new_param_filename
+if len(sys.argv) == 2 and sys.argv[1].startswith('@'):
+  param_filename = sys.argv[1][1:]
+  param_file_args = [l.strip() for l in open(param_filename, 'r').readlines()]
+else:
+  param_file_args = sys.argv[1:]
 
 emcc_py = os.path.join(os.environ['EMSCRIPTEN'], 'emcc.py')
 rtn = subprocess.call([sys.executable, emcc_py] + sys.argv[1:])

