diff --git a/Makefile.in~ b/Makefile.in
index b85ebee..53b0feb 100644
--- Makefile.in
+++ Makefile.in
@@ -15,6 +15,7 @@ VPATH		= @srcdir@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
+libdir          = @libdir@
 mandir		= @mandir@
 datarootdir	= @datarootdir@
 
@@ -368,6 +369,8 @@ install: $(PROGS)
 	$(MKDIR_P) $(DESTDIR)$(mandir)/man1
 	$(INSTALL_DATA) $(srcdir)/nasm.1 $(DESTDIR)$(mandir)/man1/nasm.1
 	$(INSTALL_DATA) $(srcdir)/ndisasm.1 $(DESTDIR)$(mandir)/man1/ndisasm.1
+	$(MKDIR_P) $(DESTDIR)$(libdir)/
+	$(INSTALL_DATA) $(srcdir)/libnasm.a $(DESTDIR)$(libdir)/
 
 clean:
 	for d in . $(SUBDIRS) $(XSUBDIRS); do \
