diff --git Makefile Makefile
index b89664c..a92476b 100644
--- Makefile
+++ Makefile
@@ -41,7 +41,6 @@ DIRS = \
 	scripts \
 	temporal \
 	doc \
-	gui \
 	visualization \
 	locale \
 	man \
diff --git display/d.where/main.c display/d.where/main.c
index a72b63a..951bc70 100644
--- display/d.where/main.c
+++ display/d.where/main.c
@@ -19,6 +19,7 @@
  *               for details.
  *
  *****************************************************************************/
+#include <stdlib.h>
 #include <string.h>
 #include <grass/gis.h>
 #include <grass/gprojects.h>
diff --git include/Make/Rules.make include/Make/Rules.make
index 071908fb3..443cb9b88 100644
--- include/Make/Rules.make
+++ include/Make/Rules.make
@@ -44,7 +44,7 @@ run_grass = \
 	PYTHONPATH="$(GRASS_PYTHONPATH)" \
 	$(LD_LIBRARY_PATH_VAR)="$(BIN):$(GISBASE)/bin:$(GISBASE)/scripts:$(ARCH_LIBDIR):$(BASE_LIBDIR):$($(LD_LIBRARY_PATH_VAR))" \
 	LC_ALL=C LANG=C LANGUAGE=C \
-	$(1)
+	$(CROSSCOMPILING_EMULATOR) $(1)
 
 # default clean rules
 clean:
diff --git lib/gis/env.c lib/gis/env.c
index 6518ff7..5f36430 100644
--- lib/gis/env.c
+++ lib/gis/env.c
@@ -115,8 +115,8 @@ void G__read_gisrc_env(void)
 }
 
 static void parse_env(FILE *fd, int loc)
-{    
-    char buf[200];
+{
+    char buf[GPATH_MAX + strlen("GISDBASE: \0\n")];
     char *name;
     char *value;
 
diff --git lib/python/Makefile lib/python/Makefile
index 2b7fb63..b0d7e73 100644
--- lib/python/Makefile
+++ lib/python/Makefile
@@ -5,7 +5,7 @@ include $(MODULE_TOPDIR)/include/Make/Python.make
 
 PYDIR = $(ETC)/python/grass
 
-SUBDIRS = exceptions script ctypes temporal pygrass pydispatch imaging gunittest
+SUBDIRS = exceptions script temporal pygrass pydispatch imaging gunittest
 
 default: $(PYDIR)/__init__.py
 	$(MAKE) subdirs
diff --git lib/raster/vrt.c lib/raster/vrt.c
index 10ae1ff..80e54a8 100644
--- lib/raster/vrt.c
+++ lib/raster/vrt.c
@@ -18,7 +18,7 @@
 
 #include "R.h"
 
-int cmp_wnd(const void *a, const void *b)
+static int cmp_wnd(const void *a, const void *b)
 {
     struct Cell_head *cellhda = &((struct tileinfo *) a)->cellhd;
     struct Cell_head *cellhdb = &((struct tileinfo *) b)->cellhd;
diff --git raster/r.mapcalc/map.c raster/r.mapcalc/map.c
index eedf19c..68ec6ec 100644
--- raster/r.mapcalc/map.c
+++ raster/r.mapcalc/map.c
@@ -23,7 +23,6 @@
 /****************************************************************************/
 
 static void prepare_region_from_maps(expression **, int, int);
-int columns;
 struct Cell_head current_region2;
 
 void setup_region(void)
