diff --git Makefile Makefile
--- Makefile
+++ Makefile
@@ -41,7 +41,6 @@ DIRS = \
 	scripts \
 	temporal \
 	doc \
-	gui \
 	visualization \
 	locale \
 	man \
diff --git include/Make/Rules.make include/Make/Rules.make
--- include/Make/Rules.make
+++ include/Make/Rules.make
@@ -44,7 +44,7 @@ run_grass = \
 	PYTHONPATH="$(GRASS_PYTHONPATH)" \
 	$(LD_LIBRARY_PATH_VAR)="$(BIN):$(GISBASE)/bin:$(GISBASE)/scripts:$(ARCH_LIBDIR):$(BASE_LIBDIR):$($(LD_LIBRARY_PATH_VAR))" \
 	LC_ALL=C LANG=C LANGUAGE=C \
-	$(1)
+	$(CROSSCOMPILING_EMULATOR) $(1)
 
 # default clean rules
 clean:
diff --git python/Makefile python/Makefile
--- python/Makefile
+++ python/Makefile
@@ -3,8 +3,7 @@
 include $(MODULE_TOPDIR)/include/Make/Vars.make
 
 SUBDIRS = \
-	grass \
-	libgrass_interface_generator
+	grass
 
 include $(MODULE_TOPDIR)/include/Make/Dir.make
 
diff --git lib/raster/vrt.c lib/raster/vrt.c
--- lib/raster/vrt.c
+++ lib/raster/vrt.c
@@ -18,7 +18,7 @@
 
 #include "R.h"
 
-int cmp_wnd(const void *a, const void *b)
+static int cmp_wnd(const void *a, const void *b)
 {
     struct Cell_head *cellhda = &((struct tileinfo *) a)->cellhd;
     struct Cell_head *cellhdb = &((struct tileinfo *) b)->cellhd;
diff --git raster/r.mapcalc/map.c raster/r.mapcalc/map.c
--- raster/r.mapcalc/map.c
+++ raster/r.mapcalc/map.c
@@ -23,7 +23,6 @@
 /****************************************************************************/
 
 static void prepare_region_from_maps(expression **, int, int);
-int columns;
 struct Cell_head current_region2;
 
 void setup_region(void)
