--- Makefile.orig	2021-07-02 22:21:22.263025540 +0000
+++ Makefile	2021-07-02 22:22:23.487179060 +0000
@@ -12,21 +12,6 @@
 # in the file LICENSE.
 # ------------------------------------------------------------------
 
-SHELL=/bin/sh
-
-# To assist in cross-compiling
-CC=gcc
-AR=ar
-RANLIB=ranlib
-LDFLAGS=
-
-BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
-
-# Where you want it installed when you do 'make install'
-PREFIX=/usr/local
-
-
 OBJS= blocksort.o  \
       huffman.o    \
       crctable.o   \
